src/Features/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -95,11 +96,6 @@ internal abstract class AbstractPullDiagnosticHandler<TDiagnosticsParams, TRepor
 
         protected abstract TReturn? CreateReturn(BufferedProgress<TReport> progress);
 
-        /// <summary>
-        /// Produce the diagnostics for the specified document.
-        /// </summary>
-        protected abstract Task<ImmutableArray<DiagnosticData>> GetDiagnosticsAsync(RequestContext context, Document document, DiagnosticMode diagnosticMode, CancellationToken cancellationToken);
```

Comment:
this moved into the implementation of each IDiagnosticSource (the method to get diagnostics for documents vs. projects is different).